blobstore/src/main/java/org/jclouds/blobstore/internal/BaseBlobStore.java

```diff
@@ -275,4 +285,20 @@ public String copyBlob(String fromContainer, String fromName, String toContainer
          Closeables2.closeQuietly(is);
       }
    }
+
+   // TODO: parallel uploads
+   @Beta
+   protected String putMultipartBlob(String container, Blob blob, PutOptions overrides) {
+      MultipartUpload mpu = initiateMultipartUpload(container, blob.getMetadata());
+      List<MultipartPart> parts = Lists.newArrayList();
+      long contentLength = blob.getMetadata().getContentMetadata().getContentLength();
+      long partSize = getMaximumMultipartPartSize();  // TODO: optimal?
```

Comment:
We need a better strategy here -- we should pick a combination of minimum part size, maximum part size, and number of parts.  A good combination will do less work when encountering network errors and allow better use of the uplink via parallel uploads.